test/integration/apiserver/anonymous/anonymous_test.go

```diff
@@ -0,0 +1,229 @@
+
+var (
+	defaultRole = &rbacv1.Role{
+		TypeMeta:   metav1.TypeMeta{APIVersion: "rbac.authorization.k8s.io/v1", Kind: "Role"},
+		ObjectMeta: metav1.ObjectMeta{Name: defaultRBACRoleName},
+		Rules: []rbacv1.PolicyRule{
+			{
+				Verbs:         []string{"list"},
+				Resources:     []string{"pods"},
+				APIGroups:     []string{""},
+				ResourceNames: []string{},
+			},
+		},
+	}
+	defaultRoleBinding = &rbacv1.RoleBinding{
+		TypeMeta:   metav1.TypeMeta{APIVersion: "rbac.authorization.k8s.io/v1", Kind: "RoleBinding"},
+		ObjectMeta: metav1.ObjectMeta{Name: defaultRBACRoleBindingName},
+		Subjects: []rbacv1.Subject{
+			{
+				APIGroup: rbac.GroupName,
+				Kind:     rbacv1.UserKind,
+				Name:     "system:anonymous",
+			},
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: rbac.GroupName,
+			Kind:     "Role",
+			Name:     defaultRBACRoleName,
+		},
+	}
+)
+
+func TestStructuredAuthenticationConfig(t *testing.T) {
+	t.Log("Testing anonymous authenticator with authentication config")
+
+	featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.StructuredAuthenticationConfiguration, true)
+	featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.AnonymousAuthConfigurableEndpoints, true)
+
+	testCases := []struct {
+		desc            string
+		authConfig      string
+		additionalFlags []string
+		assertErrFn     func(t *testing.T, errorToCheck error)
+		assertFn        func(t *testing.T, server kubeapiserverapptesting.TestServer)
+	}{
+		{
+			desc: "valid config no conditions",
+			authConfig: `
+apiVersion: apiserver.config.k8s.io/v1beta1
+kind: AuthenticationConfiguration
+anonymous:
+  enabled: true
+`,
+			assertErrFn: func(t *testing.T, errorToCheck error) {
+				require.NoError(t, errorToCheck)
+			},
+			assertFn: func(t *testing.T, server kubeapiserverapptesting.TestServer) {
+				configureRBAC(t, server)
+
+				client := insecureHTTPClient(t)
+
+				resp, err := client.Get(server.ClientConfig.Host + "/api/v1/namespaces/default/pods")
+				require.NoError(t, err)
+				defer resp.Body.Close() //nolint:errcheck
+				require.Equal(t, 200, resp.StatusCode)
+
+				resp, err = client.Get(server.ClientConfig.Host + "/livez")
+				require.NoError(t, err)
+				defer resp.Body.Close() //nolint:errcheck
+				require.Equal(t, 200, resp.StatusCode)
+
+				resp, err = client.Get(server.ClientConfig.Host + "/healthz")
+				require.NoError(t, err)
+				defer resp.Body.Close() //nolint:errcheck
+				require.Equal(t, 200, resp.StatusCode)
+
+			},
+			additionalFlags: nil,
+		},
+		{
+			desc: "valid config with conditions",
+			authConfig: `
+apiVersion: apiserver.config.k8s.io/v1beta1
+kind: AuthenticationConfiguration
+anonymous:
+  enabled: true
+  conditions:
+  - path: "/livez"
+`,
+			assertErrFn: func(t *testing.T, errorToCheck error) {
+				require.NoError(t, errorToCheck)
+			},
+			assertFn: func(t *testing.T, server kubeapiserverapptesting.TestServer) {
+				configureRBAC(t, server)
```

Comment:
the rbac roles aren't required for this, right? and the 401 indicates authn rejected before we got to authz?

Reply:
I wanted to indicate that despite the RBAC existing for it you should not be able to list pods in the default namespace.

Reply:
I think that's mostly noise... we're expecting a 401, anything else (a 200 or a 403) is a failure, regardless of why

Reply:
Removed. Thanks!